arangod/IResearch/IResearchInvertedIndex.h

```diff
@@ -27,29 +27,29 @@
 #include "Indexes/Index.h"
 #include "Indexes/IndexIterator.h"
 #include "Indexes/IndexFactory.h"
-#include "IResearchCommon.h"
-#include "IResearchDataStore.h"
-#include "IResearchInvertedIndexMeta.h"
+#include "IResearch/IResearchCommon.h"
+#include "IResearch/IResearchDataStore.h"
+#include "IResearch/IResearchInvertedIndexMeta.h"
+
 #include "search/boolean_filter.hpp"
 #include "search/bitset_doc_iterator.hpp"
 #include "search/score.hpp"
 #include "search/conjunction.hpp"
 #include "search/cost.hpp"
 #include <search/proxy_filter.hpp>
 
-namespace arangodb {
-namespace iresearch {
+namespace arangodb::iresearch {
 
 class IResearchInvertedIndex : public IResearchDataStore {
  public:
-  explicit IResearchInvertedIndex(IndexId iid, LogicalCollection& collection);
+  using IResearchDataStore::IResearchDataStore;
 
   void toVelocyPack(ArangodServer& server, TRI_vocbase_t const* defaultVocbase,
                     velocypack::Builder& builder,
                     bool writeAnalyzerDefinition) const;
 
-  std::string const& getDbName() const noexcept {
-    return _collection.vocbase().name();
+  decltype(auto) getDbName() const noexcept {
```

Comment:
Does this also work?
```suggestion
  auto getDbName() const noexcept {
```

Reply:
Not same, if `name()` return `const&` it will be different

Reply:
So yes it works, but can make copy -> not noexcept